src/service/github-allocator-registry/github-allocator-registry.service.ts

```diff
@@ -0,0 +1,146 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { createAppAuth } from '@octokit/auth-app';
+import { Octokit } from '@octokit/core';
+import { ConfigService } from '@nestjs/config';
+import {
+  HealthCheckError,
+  HealthIndicator,
+  HealthIndicatorResult,
+} from '@nestjs/terminus';
+import { LotusApiService } from '../lotus-api/lotus-api.service';
+import { AllocatorRegistry } from './types.github-allocator-registry';
+
+@Injectable()
+export class GitHubAllocatorRegistryService extends HealthIndicator {
+  private readonly logger = new Logger(GitHubAllocatorRegistryService.name);
+  private healthy = true;
+  private octokit: Octokit;
+
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly lotusApiService: LotusApiService,
+  ) {
+    super();
+  }
+
+  public async getHealth(): Promise<HealthIndicatorResult> {
+    const result = this.getStatus(
+      GitHubAllocatorRegistryService.name,
+      this.healthy,
+      {},
+    );
+
+    if (this.healthy) return result;
+    throw new HealthCheckError('Healthcheck failed', result);
+  }
+
+  private async getInstallationId(): Promise<number> {
+    const appOctokit = new Octokit({
+      authStrategy: createAppAuth,
+      auth: {
+        appId: this.configService.get<string>('GITHUB_APP_ID'),
+        privateKey: this.configService.get<string>('GITHUB_PRIVATE_KEY'),
+      },
+    });
+
+    const response = await appOctokit.request(
+      'GET /repos/{owner}/{repo}/installation',
+      {
+        owner: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_OWNER'),
+        repo: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_NAME'),
+        headers: {
+          'X-GitHub-Api-Version': '2022-11-28',
+        },
+      },
+    );
+
+    return response.data.id;
+  }
+
+  private async getOctokit(): Promise<Octokit> {
+    if (!this.octokit) {
+      this.octokit = new Octokit({
+        authStrategy: createAppAuth,
+        auth: {
+          appId: this.configService.get<string>('GITHUB_APP_ID'),
+          privateKey: this.configService.get<string>('GITHUB_PRIVATE_KEY'),
+          installationId: await this.getInstallationId(),
+        },
+      });
+    }
+    return this.octokit;
+  }
+
+  public async getAllocatorsRegistry(): Promise<AllocatorRegistry[]> {
+    const octokit = await this.getOctokit();
+    const response = (await octokit.request(
+      'GET /repos/{owner}/{repo}/contents/{path}',
+      {
+        owner: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_OWNER'),
+        repo: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_NAME'),
+        path: 'Allocators',
+        headers: {
+          'X-GitHub-Api-Version': '2022-11-28',
+        },
+      },
+    )) as any;
```

Comment:
I couldn't figure out how to force octokit types to be correct. What it actually returns doesn't match it's type definitions.

Reply:
Since its only used internally by this function and all exposed data have correct type - we dont care.